Can't use mutable variables inside a defer statement

**V version:** 0.1.15
**OS:** Mac OS X 10.14.5

**What did you do?**

Compile the following reproducer:

```
fn test_defer(a mut int) {
}

fn main() {
	a := 0
	defer { test_defer(mut a) }
}
```

**What did you expect to see?**

The source to compile successfully
 
**What did you see instead?**

The following generated C code fails to compile:

```c
 int main(int argc, char** argv) {
 init_consts();
 
int a= 0 ;
 
 /*
 {
 test_defer (& /*11 EXP:"int*" GOT:"int" */ a ) ;
 }
 */
  {
 test_defer (& /*11 EXP:"int*" GOT:"int" */ a ) ;
 }
 }
```

The compilation error happens because you can't have block comments inside block comments in C.

```
$ v test_defer.v
.test_defer.c:2962:47: error: extraneous ')' before ';'
 test_defer (& /*11 EXP:"int*" GOT:"int" */ a ) ;
                                              ^~
.test_defer.c:2965:3: error: expected identifier or '('
 */
  ^
.test_defer.c:2972:2: error: extraneous closing brace ('}')
 }
 ^
3 errors generated.
V panic: clang error
```

Thank you for the hard work.

--- a/compiler/parser.v
+++ b/compiler/parser.v
@@ -3273,14 +3273,18 @@ fn (p mut Parser) attribute() {
 
 fn (p mut Parser) defer_st() {
 	p.check(.key_defer)
-	// Wrap everything inside the defer block in /**/ comments, and save it in 
-	// `defer_text`. It will be inserted before every `return`. 
-	p.genln('/*') 
-	pos := p.cgen.lines.len 
 	p.check(.lcbr) 
+
+	pos := p.cgen.lines.len 
+
+	// Save everything inside the defer block to `defer_text`.
+	// It will be inserted before every `return`
 	p.genln('{') 
 	p.statements() 
 	p.cur_fn.defer_text = p.cgen.lines.right(pos).join('\n') + p.cur_fn.defer_text 
-	p.genln('*/') 
+
+	// Rollback p.cgen.lines
+	p.cgen.lines = p.cgen.lines.left(pos)
+	p.cgen.resetln('')
 }  
 

